lib/topological_inventory/orchestrator/event_manager.rb

```diff
@@ -0,0 +1,129 @@
+module TopologicalInventory
+  module Orchestrator
+    class EventManager
+      include Logging
+
+      ORCHESTRATOR_EVENT_NAME = "Orchestrator.sync".freeze
+      SYNC_EVENT_INTERVAL = 1.hour
+      SKIP_SUBSEQUENT_EVENTS_DURATION = 5.seconds
+
+      def self.run!(worker)
+        manager = new(worker)
+        manager.run!
+      end
+
+      def initialize(worker)
+        self.worker = worker
+        self.event_semaphore = Mutex.new
+        self.sync_semaphore  = Mutex.new
+        self.last_event_at = nil
+        self.publisher_sleep_time = 0
+      end
+
+      def run!
+        Thread.new { listener }
+        loop { publisher }
```

Comment:
Instead of this publishing a message to the topic, I think you could use `messaging_client.stop` which should break the loop every `SYNC_EVENT_INTERVAL`

Reply:
I found more straightforward solution, sync is invoked directly and messaging client isn't needed to stop.